hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -207,6 +215,22 @@ public synchronized int read(final byte[] b, final int off, final int len) throw
     }
     incrementReadOps();
     do {
+      if (nextReadPos >= fCursor - limit && nextReadPos <= fCursor) {
+        // data can be read from buffer.
```

Comment:
nit, add "some, possibly all the" and explain that this is determining the position of the buffer reads